opcua_dbus.c

```diff
@@ -192,9 +233,182 @@ bool dbus_temp_unpack_signal(GVariant *parameters, uint32_t *subscription_id, do
     return true;
 }
 
-void dbus_connect_g_signal(GCallback func)
+void dbus_connect_temp_g_signal(GCallback func)
 {
-    assert(NULL != dbusproxy);
+    assert(NULL != dbusproxy_temp);
     assert(NULL != func);
-    g_signal_connect(dbusproxy, "g-signal", func, NULL);
+    g_signal_connect(dbusproxy_temp, "g-signal", func, NULL);
+}
+
+void dbus_connect_ports_g_signal(GCallback func)
+{
+    assert(NULL != dbusproxy_ports);
+    assert(NULL != func);
+    g_signal_connect(dbusproxy_ports, "g-signal", func, NULL);
+}
+
+/**
+ * get_number_of_ioports:
+ * @brief Get number of I/O ports from dbus.
+ * @return TRUE if successful.
+ */
+bool dbus_get_number_of_ioports(uint32_t *inputs, uint32_t *outputs)
+{
+  assert(NULL != inputs);
+  assert(NULL != outputs);
+  assert(NULL != dbusproxy_ports);
+  gboolean ret = FALSE;
+  GError *error = NULL;
+  GVariant *ret_val = NULL;
+
+  ret_val = g_dbus_proxy_call_sync(dbusproxy_ports, "GetNbrPorts", NULL, G_DBUS_CALL_FLAGS_NONE, NO_TIMEOUT, NULL, &error);
+  if (NULL == ret_val)
+  {
+    LOG_E(
+      "%s/%s: Failed to get number of ports from D-Bus (%s)",
+      __FILE__,
+      __FUNCTION__,
+      error->message
+    );
+
+    g_error_free(error);
+    g_variant_unref(ret_val);
+    return false;
+  }
+  LOG_I("%s/%s: Got number of ports response from D-Bus!", __FILE__, __FUNCTION__);
+
+  if (ret_val) {
+    g_variant_get(ret_val, "(uu)", inputs, outputs);
+    ret = TRUE;
+    g_variant_unref(ret_val);
+  }
+
+  return ret;
```

Comment:
```suggestion
  return true;
```
Since we only have one case where things can go wrong (with return at line 276), I suggest we remove line 282 and the `ret` value, and replace this with `return true`.